PostHog/PostHogFeatureFlags.swift

```diff
@@ -49,6 +49,32 @@ class PostHogFeatureFlags {
         }
     }
 
+    private func isRecordingActive(_ featureFlags: [String: Any], _ sessionRecording: [String: Any]) -> Bool {
+        var recordingActive = true
+
+        // check for boolean flags
+        if let linkedFlag = sessionRecording["linkedFlag"] as? String,
+           let value = featureFlags[linkedFlag] as? Bool
+        {
+            recordingActive = value
+            // check for specific flag variant
+        } else if let linkedFlag = sessionRecording["linkedFlag"] as? [String: Any],
+                  let flag = linkedFlag["flag"] as? String,
+                  let variant = linkedFlag["variant"] as? String,
+                  let value = featureFlags[flag] as? String
+        {
+            recordingActive = value == variant
+            // check for multi flag variant (any)
+        } else if let linkedFlag = sessionRecording["linkedFlag"] as? String,
+                  featureFlags[linkedFlag] != nil
+        {
+            // always true if the flag exists
+            recordingActive = true
+        }
```

Comment:
I feel like this is not necessary since I have `linkedFlag` but nothing to compare so it means just active?

Reply:
i thought the feature flag response was always `Record<string, string|bool>` so I don't know if the third branch is correct

i.e. I don't know that we can say that the recording is active if the linked flag is present without inspecting its value

but given we've covered 

* it's a bool in the first branch
* and it's a string in the second

I think you don't need the third branch

* there is no linked flag - nothing to do
* there's a linked flag but it is some format we weren't expecting - nothing to do

since we default to active I agree I think the 3rd branch is unnecessary

otherwise the logic looks right to me

Reply:
The thing is, the 3rd branch is a valid response from the API.
You create a multi-variant flag, assign it to the recording config, and select `any`, the `linkedFlag` is a String, but you cannot check it since it is "any", it'd be a match for everything anyway, so I'd assume removing the branch is just ok since it will be active as well

Reply:
pushed changes and added tests